### DISCUSSION TITLE
Build>Core>KnowledgeMW build failure

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-Ed/Community#280] [Installation] [answered]
Hi Team,

CSP: GCP
Sunbirdevops- 5.1.0
I am trying to build  a knowledgeMW it fails
Build version - release-4.3.0_RC1

`Started by user Prasath Sivasubramaniyan
Checking out git https://github.com/project-sunbird/knowledge-mw-service.git into /var/lib/jenkins/workspace/Build/Core/KnowledgeMW@script/abdaf008f45c9839bd83f6267cee86246649d6c584d896c136c9aaaaabb5699a to read Jenkinsfile
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/lib/jenkins/workspace/Build/Core/KnowledgeMW@script/abdaf008f45c9839bd83f6267cee86246649d6c584d896c136c9aaaaabb5699a/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/project-sunbird/knowledge-mw-service.git # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/knowledge-mw-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/knowledge-mw-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git rev-parse origin/release-4.3.0_RC1^{commit} # timeout=10
 > git rev-parse release-4.3.0_RC1^{commit} # timeout=10
Checking out Revision c8c92fd9f48b9f85c3262c1239205db09abd24d4 (release-4.3.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f c8c92fd9f48b9f85c3262c1239205db09abd24d4 # timeout=10
Commit message: "Merge pull request #267 from keshavprasadms/release-4.3.0"
 > git rev-list --no-walk c8c92fd9f48b9f85c3262c1239205db09abd24d4 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] node
Running on Jenkins in /var/lib/jenkins/workspace/Build/Core/KnowledgeMW
[Pipeline] {
[Pipeline] ansiColor
[Pipeline] {

[Pipeline] stage
[Pipeline] { (Checkout)
[Pipeline] echo
[1m[32mFound environment variable named hub_org with value as: asia.gcr.io/upsmf-368011[0m
[Pipeline] cleanWs
[WS-CLEANUP] Deleting project workspace...
[WS-CLEANUP] Deferred wipeout is used...
[WS-CLEANUP] done
[Pipeline] checkout
The recommended git tool is: NONE
No credentials specified
Cloning the remote Git repository
Cloning repository https://github.com/project-sunbird/knowledge-mw-service.git
 > git init /var/lib/jenkins/workspace/Build/Core/KnowledgeMW # timeout=10
Fetching upstream changes from https://github.com/project-sunbird/knowledge-mw-service.git
 > git --version # timeout=10
 > git --version # 'git version 2.17.1'
 > git fetch --tags --progress -- https://github.com/project-sunbird/knowledge-mw-service.git +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/project-sunbird/knowledge-mw-service.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
Avoid second fetch
 > git rev-parse origin/release-4.3.0_RC1^{commit} # timeout=10
 > git rev-parse release-4.3.0_RC1^{commit} # timeout=10
Checking out Revision c8c92fd9f48b9f85c3262c1239205db09abd24d4 (release-4.3.0_RC1)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f c8c92fd9f48b9f85c3262c1239205db09abd24d4 # timeout=10
Commit message: "Merge pull request #267 from keshavprasadms/release-4.3.0"
[Pipeline] sh
+ git rev-parse --short HEAD
[Pipeline] sh
+ echo release-4.3.0_RC1_c8c92fd_10
[Pipeline] echo
build_tag: release-4.3.0_RC1_c8c92fd_10
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Build)
[Pipeline] echo
Environment will be : build
[Pipeline] sh
+ git submodule update --init
Submodule 'src/libs' (https://github.com/project-sunbird/sunbird-js-utils) registered for path 'src/libs'
Cloning into '/var/lib/jenkins/workspace/Build/Core/KnowledgeMW/src/libs'...
Submodule path 'src/libs': checked out 'f9580a929511b7c735dca3e3b7cd17f955616d19'
[Pipeline] sh
+ chmod 777 build.sh
[Pipeline] sh
+ ./build.sh release-4.3.0_RC1_c8c92fd_10 built-in asia.gcr.io/upsmf-368011
--2022-12-01 14:47:23--  https://www.imagemagick.org/download/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz
Resolving www.imagemagick.org (www.imagemagick.org)... 50.251.58.13
Connecting to www.imagemagick.org (www.imagemagick.org)|50.251.58.13|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://download.imagemagick.org/archive/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz [following]
--2022-12-01 14:47:24--  https://download.imagemagick.org/archive/binaries/ImageMagick-i386-pc-solaris2.11.tar.gz
Resolving download.imagemagick.org (download.imagemagick.org)... 50.251.58.13
Connecting to download.imagemagick.org (download.imagemagick.org)|50.251.58.13|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 13199634 (13M) [application/x-gzip]
Saving to: ‘ImageMagick-i386-pc-solaris2.11.tar.gz’

     0K .......... .......... .......... .......... ..........  0% 92.4K 2m19s
    50K .......... .......... .......... .......... ..........  0%  168K 1m47s
   100K .......... .......... .......... .......... ..........  1%  149K 1m40s
   150K .......... .......... .......... .......... ..........  1% 69.6K 2m0s
   200K .......... .......... .......... .......... ..........  1%  240K 1m46s
   250K .......... .......... .......... .......... ..........  2%  181K 1m40s
   300K .......... .......... .......... .......... ..........  2%  114K 1m41s
   350K .......... .......... .......... .......... ..........  3%  110K 1m42s
   400K .......... .......... .......... .......... ..........  3%  113K 1m43s
   450K .......... .......... .......... .......... ..........  3%  112K 1m43s
   500K .......... .......... .......... .......... ..........  4%  114K 1m43s
   550K .......... .......... .......... .......... ..........  4%  181K 1m40s
   600K .......... .......... .......... .......... ..........  5%  123K 99s
   650K .......... .......... .......... .......... ..........  5%  116K 99s
   700K .......... .......... .......... .......... ..........  5%  181K 97s
   750K .......... .......... .......... .......... ..........  6%  126K 96s
   800K .......... .......... .......... .......... ..........  6%  218K 94s
   850K .......... .......... .......... .......... ..........  6%  120K 94s
   900K .......... .......... .......... .......... ..........  7%  190K 92s
   950K .......... .......... .......... .......... ..........  7%  190K 90s
  1000K .......... .......... .......... .......... ..........  8%  215K 88s
  1050K .......... .......... .......... .......... ..........  8%  219K 86s
  1100K .......... .......... .......... .......... ..........  8%  202K 84s
  1150K .......... .......... .......... .......... ..........  9%  240K 83s
  1200K .......... .......... .......... .......... ..........  9%  271K 81s
  1250K .......... .......... .......... .......... .......... 10%  287K 79s
  1300K .......... .......... .......... .......... .......... 10%  109K 79s
  1350K .......... .......... .......... .......... .......... 10%  220M 76s
  1400K .......... .......... .......... .......... .......... 11% 80.9K 78s
  1450K .......... .......... .......... .......... .......... 11%  871K 76s
  1500K .......... .......... .......... .......... .......... 12%  322K 74s
  1550K .......... .......... .......... .......... .......... 12%  218K 73s
  1600K .......... .......... .......... .......... .......... 12%  187K 72s
  1650K .......... .......... .......... .......... .......... 13%  224K 71s
  1700K .......... .......... .......... .......... .......... 13%  203K 71s
  1750K .......... .......... .......... .......... .......... 13%  241K 70s
  1800K .......... .......... .......... .......... .......... 14%  233K 69s
  1850K .......... .......... .......... .......... .......... 14%  201K 68s
  1900K .......... .......... .......... .......... .......... 15%  242K 67s
  1950K .......... .......... .......... .......... .......... 15%  194K 67s
  2000K .......... .......... .......... .......... .......... 15%  241K 66s
  2050K .......... .......... .......... .......... .......... 16%  228K 65s
  2100K .......... .......... .......... .......... .......... 16%  184K 65s
  2150K .......... .......... .......... .......... .......... 17%  220K 64s
  2200K .......... .......... .......... .......... .......... 17%  172K 64s
  2250K .......... .......... .......... .......... .......... 17%  203K 63s
  2300K .......... .......... .......... .......... .......... 18%  223K 62s
  2350K .......... .......... .......... .......... .......... 18%  194K 62s
  2400K .......... .......... .......... .......... .......... 19%  231K 61s
  2450K .......... .......... .......... .......... .......... 19%  201K 61s
  2500K .......... .......... .......... .......... .......... 19%  200K 60s
  2550K .......... .......... .......... .......... .......... 20%  229K 60s
  2600K .......... .......... .......... .......... .......... 20%  201K 59s
  2650K .......... .......... .......... .......... .......... 20%  238K 59s
  2700K .......... .......... .......... .......... .......... 21%  206K 58s
  2750K .......... .......... .......... .......... .......... 21%  225K 58s
  2800K .......... .......... .......... .......... .......... 22%  109K 58s
  2850K .......... .......... .......... .......... .......... 22%  337K 57s
  2900K .......... .......... .......... .......... .......... 22%  686K 56s
  2950K .......... .......... .......... .......... .......... 23%  261K 56s
  3000K .......... .......... .......... .......... .......... 23%  275K 55s
  3050K .......... .......... .......... .......... .......... 24%  324K 54s
  3100K .......... .......... .......... .......... .......... 24%  234K 54s
  3150K .......... .......... .......... .......... .......... 24%  303K 53s
  3200K .......... .......... .......... .......... .......... 25%  234K 53s
  3250K .......... .......... .......... .......... .......... 25%  476K 52s
  3300K .......... .......... .......... .......... .......... 25%  331K 51s
  3350K .......... .......... .......... .......... .......... 26%  260K 51s
  3400K .......... .......... .......... .......... .......... 26%  358K 50s
  3450K .......... .......... .......... .......... .......... 27%  272K 50s
  3500K .......... .......... .......... .......... .......... 27%  361K 49s
  3550K .......... .......... .......... .......... .......... 27%  382K 49s
  3600K .......... .......... .......... .......... .......... 28%  299K 48s
  3650K .......... .......... .......... .......... .......... 28%  347K 48s
  3700K .......... .......... .......... .......... .......... 29%  359K 47s
  3750K .......... .......... .......... .......... .......... 29%  368K 47s
  3800K .......... .......... .......... .......... .......... 29%  300K 46s
  3850K .......... .......... .......... .......... .......... 30%  375K 46s
  3900K .......... .......... .......... .......... .......... 30%  473K 45s
  3950K .......... .......... .......... .......... .......... 31%  243K 45s
  4000K .......... .......... .......... .......... .......... 31%  390K 44s
  4050K .......... .......... .......... .......... .......... 31%  489K 44s
  4100K .......... .......... .......... .......... .......... 32%  231K 43s
  4150K .......... .......... .......... .......... .......... 32%  461K 43s
  4200K .......... .......... .......... .......... .......... 32%  280K 42s
  4250K .......... .......... .......... .......... .......... 33%  249K 42s
  4300K .......... .......... .......... .......... .......... 33%  440K 41s
  4350K .......... .......... .......... .......... .......... 34%  254K 41s
  4400K .......... .......... .......... .......... .......... 34%  279K 41s
  4450K .......... .......... .......... .......... .......... 34%  284K 40s
  4500K .......... .......... .......... .......... .......... 35%  269K 40s
  4550K .......... .......... .......... .......... .......... 35%  299K 40s
  4600K .......... .......... .......... .......... .......... 36%  253K 39s
  4650K .......... .......... .......... .......... .......... 36%  347K 39s
  4700K .......... .......... .......... .......... .......... 36%  321K 39s
  4750K .......... .......... .......... .......... .......... 37%  430K 38s
  4800K .......... .......... .......... .......... .......... 37%  385K 38s
  4850K .......... .......... .......... .......... .......... 38%  346K 37s
  4900K .......... .......... .......... .......... .......... 38%  349K 37s
  4950K .......... .......... .......... .......... .......... 38%  347K 37s
  5000K .......... .......... .......... .......... .......... 39%  408K 36s
  5050K .......... .......... .......... .......... .......... 39%  336K 36s
  5100K .......... .......... .......... .......... .......... 39%  382K 35s
  5150K .......... .......... .......... .......... .......... 40%  354K 35s
  5200K .......... .......... .......... .......... .......... 40%  326K 35s
  5250K .......... .......... .......... .......... .......... 41%  361K 34s
  5300K .......... .......... .......... .......... .......... 41%  387K 34s
  5350K .......... .......... .......... .......... .......... 41%  284K 34s
  5400K .......... .......... .......... .......... .......... 42%  406K 33s
  5450K .......... .......... .......... .......... .......... 42%  283K 33s
  5500K .......... .......... .......... .......... .......... 43% 86.0K 33s
  5550K .......... .......... .......... .......... .......... 43%  261M 33s
  5600K .......... .......... .......... .......... .......... 43%  247K 33s
  5650K .......... .......... .......... .......... .......... 44%  209M 32s
  5700K .......... .......... .......... .......... .......... 44%  370K 32s
  5750K .......... .......... .......... .......... .......... 44%  226M 31s
  5800K .......... .......... .......... .......... .......... 45%  646K 31s
  5850K .......... .......... .......... .......... .......... 45%  279K 31s
  5900K .......... .......... .......... .......... .......... 46%  365K 30s
  5950K .......... .......... .......... .......... .......... 46%  271K 30s
  6000K .......... .......... .......... .......... .......... 46%  280K 30s
  6050K .......... .......... .......... .......... .......... 47%  262K 29s
  6100K .......... .......... .......... .......... .......... 47%  332K 29s
  6150K .......... .......... .......... .......... .......... 48%  383K 29s
  6200K .......... .......... .......... .......... .......... 48%  250K 29s
  6250K .......... .......... .......... .......... .......... 48%  304K 28s
  6300K .......... .......... .......... .......... .......... 49%  424K 28s
  6350K .......... .......... .......... .......... .......... 49%  297K 28s
  6400K .......... .......... .......... .......... .......... 50%  274K 28s
  6450K .......... .......... .......... .......... .......... 50%  315K 27s
  6500K .......... .......... .......... .......... .......... 50%  479K 27s
  6550K .......... .......... .......... .......... .......... 51%  238K 27s
  6600K .......... .......... .......... .......... .......... 51%  302K 27s
  6650K .......... .......... .......... .......... .......... 51%  542K 26s
  6700K .......... .......... .......... .......... .......... 52%  258K 26s
  6750K .......... .......... .......... .......... .......... 52%  304K 26s
  6800K .......... .......... .......... .......... .......... 53%  426K 25s
  6850K .......... .......... .......... .......... .......... 53%  295K 25s
  6900K .......... .......... .......... .......... .......... 53%  266K 25s
  6950K .......... .......... .......... .......... .......... 54%  323K 25s
  7000K .......... .......... .......... .......... .......... 54%  484K 24s
  7050K .......... .......... .......... .......... .......... 55%  240K 24s
  7100K .......... .......... .......... .......... .......... 55%  289K 24s
  7150K .......... .......... .......... .......... .......... 55%  633K 24s
  7200K .......... .......... .......... .......... .......... 56%  250K 23s
  7250K .......... .......... .......... .......... .......... 56%  299K 23s
  7300K .......... .......... .......... .......... .......... 57%  619K 23s
  7350K .......... .......... .......... .......... .......... 57%  255K 23s
  7400K .......... .......... .......... .......... .......... 57%  372K 22s
  7450K .......... .......... .......... .......... .......... 58%  363K 22s
  7500K .......... .......... .......... .......... .......... 58%  317K 22s
  7550K .......... .......... .......... .......... .......... 58%  437K 22s
  7600K .......... .......... .......... .......... .......... 59%  366K 21s
  7650K .......... .......... .......... .......... .......... 59%  272K 21s
  7700K .......... .......... .......... .......... .......... 60%  429K 21s
  7750K .......... .......... .......... .......... .......... 60%  266K 21s
  7800K .......... .......... .......... .......... .......... 60%  540K 20s
  7850K .......... .......... .......... .......... .......... 61%  287K 20s
  7900K .......... .......... .......... .......... .......... 61%  377K 20s
  7950K .......... .......... .......... .......... .......... 62%  375K 20s
  8000K .......... .......... .......... .......... .......... 62%  345K 19s
  8050K .......... .......... .......... .......... .......... 62%  365K 19s
  8100K .......... .......... .......... .......... .......... 63%  288K 19s
  8150K .......... .......... .......... .......... .......... 63%  365K 19s
  8200K .......... .......... .......... .......... .......... 64%  130K 19s
  8250K .......... .......... .......... .......... .......... 64%  601K 18s
  8300K .......... .......... .......... .......... .......... 64%  427K 18s
  8350K .......... .......... .......... .......... .......... 65%  307K 18s
  8400K .......... .......... .......... .......... .......... 65%  194M 18s
  8450K .......... .......... .......... .......... .......... 65%  488K 17s
  8500K .......... .......... .......... .......... .......... 66%  249K 17s
  8550K .......... .......... .......... .......... .......... 66%  295K 17s
  8600K .......... .......... .......... .......... .......... 67%  231K 17s
  8650K .......... .......... .......... .......... .......... 67%  348K 17s
  8700K .......... .......... .......... .......... .......... 67%  260K 16s
  8750K .......... .......... .......... .......... .......... 68%  301K 16s
  8800K .......... .......... .......... .......... .......... 68%  298K 16s
  8850K .......... .......... .......... .......... .......... 69%  268K 16s
  8900K .......... .......... .......... .......... .......... 69%  336K 16s
  8950K .......... .......... .......... .......... .......... 69%  299K 15s
  9000K .......... .......... .......... .......... .......... 70%  350K 15s
  9050K .......... .......... .......... .......... .......... 70%  351K 15s
  9100K .......... .......... .......... .......... .......... 70%  275K 15s
  9150K .......... .......... .......... .......... .......... 71%  299K 14s
  9200K .......... .......... .......... .......... .......... 71%  284K 14s
  9250K .......... .......... .......... .......... .......... 72%  336K 14s
  9300K .......... .......... .......... .......... .......... 72%  309K 14s
  9350K .......... .......... .......... .......... .......... 72%  274K 14s
  9400K .......... .......... .......... .......... .......... 73%  299K 13s
  9450K .......... .......... .......... .......... .......... 73%  318K 13s
  9500K .......... .......... .......... .......... .......... 74%  321K 13s
  9550K .......... .......... .......... .......... .......... 74%  336K 13s
  9600K .......... .......... .......... .......... .......... 74%  246K 13s
  9650K .......... .......... .......... .......... .......... 75%  380K 12s
  9700K .......... .......... .......... .......... .......... 75%  314K 12s
  9750K .......... .......... .......... .......... .......... 76%  294K 12s
  9800K .......... .......... .......... .......... .......... 76%  333K 12s
  9850K .......... .......... .......... .......... .......... 76%  318K 12s
  9900K .......... .......... .......... .......... .......... 77%  349K 11s
  9950K .......... .......... .......... .......... .......... 77%  336K 11s
 10000K .......... .......... .......... .......... .......... 77%  317K 11s
 10050K .......... .......... .......... .......... .......... 78%  375K 11s
 10100K .......... .......... .......... .......... .......... 78%  321K 11s
 10150K .......... .......... .......... .......... .......... 79%  391K 10s
 10200K .......... .......... .......... .......... .......... 79%  312K 10s
 10250K .......... .......... .......... .......... .......... 79%  397K 10s
 10300K .......... .......... .......... .......... .......... 80%  381K 10s
 10350K .......... .......... .......... .......... .......... 80%  405K 10s
 10400K .......... .......... .......... .......... .......... 81%  370K 9s
 10450K .......... .......... .......... .......... .......... 81%  338K 9s
 10500K .......... .......... .......... .......... .......... 81%  510K 9s
 10550K .......... .......... .......... .......... .......... 82%  404K 9s
 10600K .......... .......... .......... .......... .......... 82%  491K 8s
 10650K .......... .......... .......... .......... .......... 83%  393K 8s
 10700K .......... .......... .......... .......... .......... 83%  512K 8s
 10750K .......... .......... .......... .......... .......... 83%  401K 8s
 10800K .......... .......... .......... .......... .......... 84%  531K 8s
 10850K .......... .......... .......... .......... .......... 84%  389K 7s
 10900K .......... .......... .......... .......... .......... 84%  675K 7s
 10950K .......... .......... .......... .......... .......... 85%  543K 7s
 11000K .......... .......... .......... .......... .......... 85%  527K 7s
 11050K .......... .......... .......... .......... .......... 86%  792K 7s
 11100K .......... .......... .......... .......... .......... 86%  391K 6s
 11150K .......... .......... .......... .......... .......... 86%  747K 6s
 11200K .......... .......... .......... .......... .......... 87%  514K 6s
 11250K .......... .......... .......... .......... .......... 87%  535K 6s
 11300K .......... .......... .......... .......... .......... 88%  876K 6s
 11350K .......... .......... .......... .......... .......... 88%  527K 5s
 11400K .......... .......... .......... .......... .......... 88%  669K 5s
 11450K .......... .......... .......... .......... .......... 89%  883K 5s
 11500K .......... .......... .......... .......... .......... 89%  685K 5s
 11550K .......... .......... .......... .......... .......... 89%  578K 5s
 11600K .......... .......... .......... .......... .......... 90%  808K 4s
 11650K .......... .......... .......... .......... .......... 90%  695K 4s
 11700K .......... .......... .......... .......... .......... 91%  829K 4s
 11750K .......... .......... .......... .......... .......... 91%  617K 4s
 11800K .......... .......... .......... .......... .......... 91% 1.98M 4s
 11850K .......... .......... .......... .......... .......... 92%  457K 4s
 11900K .......... .......... .......... .......... .......... 92% 1.05M 3s
 11950K .......... .......... .......... .......... .......... 93%  640K 3s
 12000K .......... .......... .......... .......... .......... 93%  790K 3s
 12050K .......... .......... .......... .......... .......... 93% 1.04M 3s
 12100K .......... .......... .......... .......... .......... 94%  940K 3s
 12150K .......... .......... .......... .......... .......... 94%  534K 2s
 12200K .......... .......... .......... .......... .......... 95%  835K 2s
 12250K .......... .......... .......... .......... .......... 95% 1023K 2s
 12300K .......... .......... .......... .......... .......... 95%  882K 2s
 12350K .......... .......... .......... .......... .......... 96%  537K 2s
 12400K .......... .......... .......... .......... .......... 96% 1.34M 2s
 12450K .......... .......... .......... .......... .......... 96%  772K 1s
 12500K .......... .......... .......... .......... .......... 97%  699K 1s
 12550K .......... .......... .......... .......... .......... 97%  916K 1s
 12600K .......... .......... .......... .......... .......... 98%  764K 1s
 12650K .......... .......... .......... .......... .......... 98% 1.03M 1s
 12700K .......... .......... .......... .......... .......... 98%  853K 0s
 12750K .......... .......... .......... .......... .......... 99%  716K 0s
 12800K .......... .......... .......... .......... .......... 99%  970K 0s
 12850K .......... .......... .......... ..........           100%  947K=44s

2022-12-01 14:48:09 (295 KB/s) - ‘ImageMagick-i386-pc-solaris2.11.tar.gz’ saved [13199634/13199634]

Sending build context to Docker daemon  17.34MB

Step 1/19 : FROM circleci/node:8.11.2-stretch
 ---> 3c168695e85e
Step 2/19 : MAINTAINER "Manojvv" "manojv@ilimi.in"
 ---> Using cache
 ---> 105d7cdc3b88
Step 3/19 : USER root
 ---> Using cache
 ---> 6101febbd788
Step 4/19 : COPY src /opt/content/
 ---> Using cache
 ---> a495bac75787
Step 5/19 : WORKDIR /opt/content/
 ---> Using cache
 ---> 6b68b091435e
Step 6/19 : RUN npm install --unsafe-perm
 ---> Using cache
 ---> bc4a144c79d4
Step 7/19 : FROM node:8.11-slim
 ---> 1aa312dac871
Step 8/19 : MAINTAINER "Manojvv" "manojv@ilimi.in"
 ---> Using cache
 ---> 19635c3d05b0
Step 9/19 : RUN sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update && apt install openssl imagemagick -y     && apt-get clean     && useradd -m sunbird
 ---> Running in 0ac32b62735e
[91m
WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

[0mIgn http://deb.debian.org jessie InRelease
Get:1 http://deb.debian.org jessie Release.gpg [1652 B]
Get:2 http://deb.debian.org jessie Release [77.3 kB]
Get:3 http://security.debian.org jessie/updates InRelease [44.9 kB]
Ign http://deb.debian.org jessie Release
Get:4 http://deb.debian.org jessie/main amd64 Packages [9098 kB]
Get:5 http://security.debian.org jessie/updates/main amd64 Packages [992 kB]
Fetched 10.2 MB in 4s (2294 kB/s)
Reading package lists...
Building dependency tree...
Reading state information...
37 packages can be upgraded. Run 'apt list --upgradable' to see them.
[91mW: GPG error: http://deb.debian.org jessie Release: The following signatures were invalid: KEYEXPIRED 1668891673
[0m[91m
WARNING: apt does not have a stable CLI interface yet. Use with caution in scripts.

[0mReading package lists...
Building dependency tree...
Reading state information...
The following extra packages will be installed:
  dbus fontconfig fontconfig-config fonts-dejavu-core fonts-droid gcc-4.9-base
  ghostscript gsfonts hicolor-icon-theme ifupdown imagemagick-6.q16
  imagemagick-common iproute2 isc-dhcp-client isc-dhcp-common
  libalgorithm-c3-perl libarchive-extract-perl libatm1 libavahi-client3
  libavahi-common-data libavahi-common3 libcairo2 libcap-ng0 libcgi-fast-perl
  libcgi-pm-perl libclass-c3-perl libclass-c3-xs-perl libcpan-meta-perl
  libcroco3 libcups2 libcupsfilters1 libcupsimage2 libdata-optlist-perl
  libdata-section-perl libdatrie1 libdbus-1-3 libdjvulibre-text libdjvulibre21
  libdns-export100 libexpat1 libfcgi-perl libfftw3-double3 libfontconfig1
  libfreetype6 libgcc1 libgdbm3 libgdk-pixbuf2.0-0 libgdk-pixbuf2.0-common
  libglib2.0-0 libglib2.0-data libgomp1 libgraphite2-3 libgs9 libgs9-common
  libharfbuzz0b libijs-0.35 libilmbase6 libirs-export91 libisc-export95
  libisccfg-export90 libjasper1 libjbig0 libjbig2dec0 libjpeg62-turbo
  liblcms2-2 liblog-message-perl liblog-message-simple-perl liblqr-1-0
  libltdl7 libmagickcore-6.q16-2 libmagickcore-6.q16-2-extra
  libmagickwand-6.q16-2 libmodule-build-perl libmodule-pluggable-perl
  libmodule-signature-perl libmro-compat-perl libnetpbm10 libopenexr6
  libopenjp2-7 libpackage-constants-perl libpango-1.0-0 libpangocairo-1.0-0
  libpangoft2-1.0-0 libpaper-utils libpaper1 libparams-util-perl libpixman-1-0
  libpng12-0 libpod-latex-perl libpod-readme-perl libregexp-common-perl
  librsvg2-2 librsvg2-common libsoftware-license-perl libstdc++6
  libsub-exporter-perl libsub-install-perl libterm-ui-perl
  libtext-soundex-perl libtext-template-perl libthai-data libthai0 libtiff5
  libwmf0.2-7 libx11-6 libx11-data libxau6 libxcb-render0 libxcb-shm0 libxcb1
  libxdmcp6 libxext6 libxml2 libxrender1 libxtables10 netbase netpbm perl
  perl-base perl-modules poppler-data rename sgml-base shared-mime-info ucf
  xdg-user-dirs xml-core
Suggested packages:
  dbus-x11 ghostscript-x ppp rdnssd net-tools imagemagick-doc autotrace
  cups-bsd lpr lprng enscript ffmpeg gimp gnuplot grads graphviz groff-base
  hp2xx html2ps libwmf-bin mplayer povray radiance sane-utils texlive-base-bin
  transfig ufraw-batch xdg-utils iproute2-doc resolvconf avahi-autoipd
  cups-common libfftw3-bin libfftw3-dev libjasper-runtime liblcms2-utils
  inkscape librsvg2-bin perl-doc libterm-readline-gnu-perl
  libterm-readline-perl-perl make libb-lint-perl libcpanplus-dist-build-perl
  libcpanplus-perl libfile-checktree-perl libobject-accessor-perl
  poppler-utils fonts-japanese-mincho fonts-ipafont-mincho
  fonts-japanese-gothic fonts-ipafont-gothic fonts-arphic-ukai
  fonts-arphic-uming fonts-nanum sgml-base-doc debhelper
Recommended packages:
  libarchive-tar-perl
The following NEW packages will be installed:
  dbus fontconfig fontconfig-config fonts-dejavu-core fonts-droid ghostscript
  gsfonts hicolor-icon-theme ifupdown imagemagick imagemagick-6.q16
  imagemagick-common iproute2 isc-dhcp-client isc-dhcp-common
  libalgorithm-c3-perl libarchive-extract-perl libatm1 libavahi-client3
  libavahi-common-data libavahi-common3 libcairo2 libcap-ng0 libcgi-fast-perl
  libcgi-pm-perl libclass-c3-perl libclass-c3-xs-perl libcpan-meta-perl
  libcroco3 libcups2 libcupsfilters1 libcupsimage2 libdata-optlist-perl
  libdata-section-perl libdatrie1 libdbus-1-3 libdjvulibre-text libdjvulibre21
  libdns-export100 libexpat1 libfcgi-perl libfftw3-double3 libfontconfig1
  libfreetype6 libgdbm3 libgdk-pixbuf2.0-0 libgdk-pixbuf2.0-common
  libglib2.0-0 libglib2.0-data libgomp1 libgraphite2-3 libgs9 libgs9-common
  libharfbuzz0b libijs-0.35 libilmbase6 libirs-export91 libisc-export95
  libisccfg-export90 libjasper1 libjbig0 libjbig2dec0 libjpeg62-turbo
  liblcms2-2 liblog-message-perl liblog-message-simple-perl liblqr-1-0
  libltdl7 libmagickcore-6.q16-2 libmagickcore-6.q16-2-extra
  libmagickwand-6.q16-2 libmodule-build-perl libmodule-pluggable-perl
  libmodule-signature-perl libmro-compat-perl libnetpbm10 libopenexr6
  libopenjp2-7 libpackage-constants-perl libpango-1.0-0 libpangocairo-1.0-0
  libpangoft2-1.0-0 libpaper-utils libpaper1 libparams-util-perl libpixman-1-0
  libpng12-0 libpod-latex-perl libpod-readme-perl libregexp-common-perl
  librsvg2-2 librsvg2-common libsoftware-license-perl libsub-exporter-perl
  libsub-install-perl libterm-ui-perl libtext-soundex-perl
  libtext-template-perl libthai-data libthai0 libtiff5 libwmf0.2-7 libx11-6
  libx11-data libxau6 libxcb-render0 libxcb-shm0 libxcb1 libxdmcp6 libxext6
  libxml2 libxrender1 libxtables10 netbase netpbm perl perl-modules
  poppler-data rename sgml-base shared-mime-info ucf xdg-user-dirs xml-core
The following packages will be upgraded:
  gcc-4.9-base libgcc1 libstdc++6 openssl perl-base
5 upgraded, 124 newly installed, 0 to remove and 32 not upgraded.
Need to get 48.8 MB of archives.
After this operation, 168 MB of additional disk space will be used.
WARNING: The following packages cannot be authenticated!
  libgdbm3 libpng12-0 ucf fonts-dejavu-core fonts-droid libatm1
  libavahi-common-data libavahi-common3 libavahi-client3 libxau6 libxcb1
  libxcb-render0 libxcb-shm0 libxext6 libxrender1 libcairo2 libcap-ng0
  libcroco3 libjbig0 libdatrie1 libfftw3-double3 libijs-0.35 libilmbase6
  liblqr-1-0 libltdl7 libopenexr6 libthai-data libthai0 libpango-1.0-0
  libpangoft2-1.0-0 libpangocairo-1.0-0 libpaper1 libwmf0.2-7 poppler-data
  sgml-base iproute2 ifupdown libxtables10 netbase gsfonts hicolor-icon-theme
  libalgorithm-c3-perl libarchive-extract-perl libcgi-pm-perl libfcgi-perl
  libcgi-fast-perl libclass-c3-perl libclass-c3-xs-perl libcpan-meta-perl
  libparams-util-perl libsub-install-perl libdata-optlist-perl
  libmro-compat-perl libsub-exporter-perl libdata-section-perl
  liblog-message-perl liblog-message-simple-perl libmodule-pluggable-perl
  libnetpbm10 libpackage-constants-perl libpaper-utils libpod-latex-perl
  libregexp-common-perl libpod-readme-perl libtext-template-perl
  libsoftware-license-perl libterm-ui-perl libtext-soundex-perl netpbm rename
  shared-mime-info xdg-user-dirs xml-core
[91mE: There are problems and -y was used without --force-yes
[0mThe command '/bin/sh -c sed -i '/jessie-updates/d' /etc/apt/sources.list     && apt update && apt install openssl imagemagick -y     && apt-get clean     && useradd -m sunbird' returned a non-zero code: 100
[Pipeline] }
[Pipeline] // stage
[Pipeline] }

[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
ERROR: script returned exit code 100
Finished: FAILURE
`

**Comment by beepdot** [comment]
@vinukumar-vs Can you please check and respond on this?

**Reply by HarishGangula** [reply]
@tsprasath  This issue is fixed in [here](https://github.com/project-sunbird/knowledge-mw-service/tree/release-5.3.0). Please use this version of the code

**Comment by tsprasath** [comment]
@HarishGangula Builded `release-5.3.0` and deployed it successfully.
@keshavprasadms  Close this thread

**Comment by vinukumar-vs** [comment, marked as answer]
@tsprasath One of the dependency library is deprecated now. Hence when you install is failing. Please find the below fix we did in our latest release(not yet push to production)

[development phase - PR changes of release-5.3.0](https://github.com/project-sunbird/knowledge-mw-service/commit/7ff64528fde5f1ab852139da01011d8097422bd6#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47)

@keshavprasadms please suggest how adopters can take these changes for older versions?

Can you change the Dockerfile(as suggested in PR) & try it?


@HarishGangula @Krishnaj20

**Reply by HarishGangula** [reply]
@vinukumar-vs  The issue is resolved. [check](https://github.com/orgs/Sunbird-Ed/discussions/280#discussioncomment-4289825)
